src/main/java/com/tutorhub/repository/SubmoduleRepository.java

```diff
@@ -2,26 +2,11 @@
 
 import com.tutorhub.model.course.Submodule;
 import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.data.jpa.repository.Query;
 
 import java.util.Optional;
 
 public interface SubmoduleRepository extends JpaRepository<Submodule, Long> {
-  @Query(
-      value =
-          """
-              SELECT s.id,
-                     s.name,
-                     s.module_id,
-                     s.content_id
-              FROM submodule s LEFT JOIN module m ON s.module_id = m.id
-              LEFT JOIN course c ON m.course_id = c.id
-              WHERE s.id = :submoduleId
-              AND m.id = :moduleId
-              AND c.id = :courseId
-              """,
-      nativeQuery = true)
-  Optional<Submodule> findByCourseIdAndModuleIdAndSubmoduleId(Long courseId,
-                                                              Long moduleId,
-                                                              Long submoduleId);
+
+  Optional<Submodule> findByModule_Course_IdAndModule_IdAndId(
```

Comment:
I mean you can use `findByCourseIdAndModuleIdAndSubmoduleId(...)`, do not use underscores anywhere in Java))

JPA can handle `findByCourseIdAndModuleIdAndSubmoduleId(...)` with logic that you wrote before. This is valid name, you need just to remove `@Query(...)`

Reply:
I have tried  `findByCourseIdAndModuleIdAndSubmoduleId(...)` , it returns an error because we don't have `Course` in `Submodule`

Reply:
No property 'courseId' found for type 'Submodule'

Reply:
Does your method work?

Reply:
Method `findByModule_Course_IdAndModule_IdAndId` generates the exact same query as I wrote in @Query, but `findByCourseIdAndModuleIdAndSubmoduleId(...)` throws an error with message `No property 'courseId' found for type 'Submodule'`

Reply:
Ok, I understand. If it can not be avoided, we should use it as it is

Reply:
Ok, thanks